Simplify extend bounds tests

billdenney opened the pull request:
This simplifies the test-bounds.R file so that it uses a lot more data.frames() instead of structures.  This will make it simpler to update tests in the future.

No tests were explicitly removed.  Tests for the "err" column are skipped.  A few tests of making the data.frames were added.

mattfidler commented:
This deleted the following `theta` fixed tests:

```r
a <- fix(0,2,3)
```

and 

```r
a <- c(0,fix(2),3)
```

billdenney commented:
Good catch and thanks for the review.  I've updated the tests to include these two.  While doing the updates, testthat complained about the reference values not being available, so I added them to each of the test sets, where required..

mattfidler commented:
Thanks bill
